### PR TITLE
feat: argon2-cffi_bindings package

### PR DESCRIPTION
**Pull request by stuertz**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by stuertz** [comment]
@conda-forge/help-python-c ready for review

File: recipes/argon2_cffi_bindings/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{% set name = "argon2-cffi-bindings" %}
+{% set version = "21.2.0" %}
+{% set sha256 = "bb89ceffa6c791807d1305ceb77dbfacc5aa499891d2c55661c6459651fc39e3" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
```
Comment (xhochy) [inline recipes/argon2_cffi_bindings/meta.yaml:10]: Not needed anymore
```suggestion
```

File: recipes/argon2_cffi_bindings/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (xhochy) [inline recipes/argon2_cffi_bindings/meta.yaml:12]: Please put the sha directly here

File: recipes/argon2_cffi_bindings/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    - {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  build:
+    - cffi >=1.0.1                            # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - python                                 # [build_platform != target_platform]
+    - setuptools >=45
+    - setuptools_scm >=6.2
+    - wheel
```
Comment (xhochy) [inline recipes/argon2_cffi_bindings/meta.yaml:27]: This should go into host

File: recipes/argon2_cffi_bindings/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  host:
+    - python
+    - pip
+    - cffi >=1.0.1
+    - setuptools_scm >=6.2
+
+  run:
+    - python
+    - cffi >=1.0.1
+    - setuptools_scm >=6.2
```
Comment (xhochy) [inline recipes/argon2_cffi_bindings/meta.yaml:39]: This should be a build time only dependency
```suggestion
```

File: recipes/argon2_cffi_bindings/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+test:
+  imports:
+    - _argon2_cffi_bindings
```
Comment (xhochy) [inline recipes/argon2_cffi_bindings/meta.yaml:43]:
```suggestion
    - _argon2_cffi_bindings
  requires:
    - pip
  commands:
    - pip check
```

**Pull request merged**